packages/mco/components/disaster-recovery/app-failover-relocate/dr-policy-selector.tsx

```diff
@@ -0,0 +1,133 @@
+import { useCustomTranslation } from '@odf/shared/useCustomTranslationHook';
+import { referenceForModel } from '@odf/shared/utils';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+import { Dropdown, DropdownToggle, DropdownItem } from '@patternfly/react-core';
+import { DRPolicyModel } from '../../../models';
+import { DRPolicyKind } from '../../../types';
+import {
+  DRPolicyControlState,
+  FailoverAndRelocateState,
+  FailoverAndRelocateAction,
+  FailoverAndRelocateType,
+  DRPolicyType,
+  ModalFooterStatus,
+} from './reducer';
+
+const getDRPolicyNames = (drpcState: DRPolicyControlState[]): string[] => [
+  ...new Set(
+    drpcState.map(
+      (drpcState) => drpcState?.drPolicyControl?.spec?.drPolicyRef?.name
+    )
+  ),
+];
+
+const getDRPolicyList = (
+  drpolicyNames: string[],
+  drPolicies: DRPolicyKind[]
+): DRPolicyKind[] =>
+  drPolicies.filter((drPolicy) => drpolicyNames.includes(getName(drPolicy)));
+
+export const DRPolicySelector: React.FC<DRPolicySelectorProps> = ({
+  state,
+  dispatch,
+}) => {
+  const { t } = useCustomTranslation();
+  const [drPolicies, drPoliciesLoaded, drPoliciesLoadedError] =
+    useK8sWatchResource<DRPolicyKind[]>({
+      kind: referenceForModel(DRPolicyModel),
+      isList: true,
+    });
+  const memoizedDRPolicies = useDeepCompareMemoize(drPolicies, true);
+  const memoizedDRPCState = useDeepCompareMemoize(
+    state.drPolicyControlState,
+    true
+  );
+  const [isOpen, setOpen] = React.useState(false);
+
+  const setSelected = (selection: DRPolicyType) => {
+    dispatch({
+      type: FailoverAndRelocateType.SET_SELECTED_DR_POLICY,
+      payload: selection,
+    });
+  };
+
+  const drPolicyList = React.useMemo(
+    () =>
+      drPoliciesLoaded &&
+      !drPoliciesLoadedError &&
+      !!memoizedDRPCState.length &&
+      !!memoizedDRPolicies.length
+        ? getDRPolicyList(
+            getDRPolicyNames(memoizedDRPCState),
+            memoizedDRPolicies
+          )
+        : [],
+    [
+      memoizedDRPCState,
+      memoizedDRPolicies,
+      drPoliciesLoaded,
+      drPoliciesLoadedError,
+    ]
+  );
+
+  const dropdownItems = React.useMemo(
+    () =>
+      drPolicyList.map((drPolicy, index) => (
+        <DropdownItem
+          id={index.toString()}
+          key={getUID(drPolicy)}
+          value={getName(drPolicy)}
+          component="button"
+        >
+          {getName(drPolicy)}
+        </DropdownItem>
+      )),
+    [drPolicyList]
+  );
+
+  const onToggle = (isOpen: boolean) => {
+    setOpen(isOpen);
+  };
+
+  const onSelect = (e) => {
+    // Selecting a DRPolicy to intiate failover or relocate
+    const drPolicy: DRPolicyKind = drPolicyList[e.currentTarget.id];
+    setSelected({
+      policyName: getName(drPolicy),
+      drClusters: drPolicy?.spec?.drClusters,
+    });
+    setOpen(false);
+  };
+
+  return (
+    <>
+      <Dropdown
+        className="mco-dr-action-body__dropdown-width"
```

Comment:
```suggestion
        className="mco-dr-action-body__dropdown--width"
```
rather than `width` use `wide` or `narrow` to define the type of Modifier it is.